pylint/lint/pylinter.py

```diff
@@ -567,16 +580,10 @@ def __init__(self, options=(), reporter=None, option_groups=(), pylintrc=None):
         self._dynamic_plugins = set()
         self._error_mode = False
         self.load_provider_defaults()
-        if reporter:
-            self.set_reporter(reporter)
 
     def load_default_plugins(self):
         checkers.initialize(self)
         reporters.initialize(self)
-        # Make sure to load the default reporter, because
-        # the option has been set before the plugins had been loaded.
-        if not self.reporter:
-            self._load_reporters()
```

Comment:
I like how it already simplify the code by removing handling that is dispersed in place it should not be.

Reply:
`MultiReporter` might create some additional problems for us. Have not looked into it fully, but having a reporter "that's not actually a reporter" on `PyLinter.reporter` might be troublesome..